src/panels/lovelace/components/hui-conditional.ts

```diff
@@ -0,0 +1,154 @@
+  }
+
+  public setConditionalConfig(config) {
+    if (
+      !config.conditions ||
+      !Array.isArray(config.conditions) ||
+      !config.conditions.every((c) => c.entity && (c.state || c.state_not)) ||
+      (!config.card && (!config.elements || !Array.isArray(config.elements)))
+    ) {
+      throw new Error("Error in conditional configuration.");
+    }
+
+    this._conditionalConfig = config;
+
+    // set style (overridable)
+    // LitElement static styles can't work since we want to allow deriving classes to modify it
+    const styleElement: HTMLStyleElement = document.createElement("style");
+    styleElement.innerHTML = this.getCustomStyle();
+    this.shadowRoot!.appendChild(styleElement);
+
+    this.removeItems();
+
+    this.createItems();
+
+    if (this._hass) {
+      this.hass = this._hass;
+    }
+  }
+
+  protected getCustomStyle(): string {
+    return "";
+  }
+
+  protected _createHuiElement(
+    _elementConfig: LovelaceElementConfig
+  ): LovelaceElement {
+    return (null as unknown) as LovelaceElement;
+  }
+
+  private createItems() {
+    if (this._conditionalConfig!.card) {
+      this._card = createCardElement(this._conditionalConfig!.card);
+    } else if (
+      this._conditionalConfig!.elements &&
+      this._conditionalConfig!.elements.length > 0
+    ) {
+      this._conditionalConfig!.elements.map(
+        (elementConfig: LovelaceElementConfig) => {
+          this._elements.push(this._createHuiElement(elementConfig));
+        }
+      );
+    }
+    // todo: add rows, etc'....
+  }
+
+  private handleItems(): boolean {
+    const conditionsMet =
+      this._conditionalConfig &&
+      this._conditionalConfig.conditions.every((c) => {
+        if (!(c.entity in this._hass!.states)) {
+          return false;
+        }
+        if (c.state) {
+          return this._hass!.states[c.entity].state === c.state;
+        }
+        return this._hass!.states[c.entity].state !== c.state_not;
+      });
+
+    if (conditionsMet) {
+      this.appendItems();
+    } else {
+      this.removeItems();
+    }
+
+    return conditionsMet as boolean;
+  }
+
+  private appendItems() {
+    if (this._card) {
+      this._card.hass = this._hass;
+      if (!this._card!.parentNode) {
+        this.shadowRoot!.appendChild(this._card);
+      }
+    } else if (this._elements.length > 0) {
+      this._elements.map((el: LovelaceElement) => {
+        el.hass = this._hass;
+        if (!el.parentNode) {
+          this.shadowRoot!.appendChild(el);
+        }
+      });
+    }
+  }
+
+  private removeItems() {
+    if (this._card && this._card.parentNode) {
+      this.shadowRoot!.removeChild(this._card);
+    } else if (this._elements.length > 0) {
+      this._elements.map((el: LovelaceElement) => {
+        if (el.parentNode) {
+          this.shadowRoot!.removeChild(el);
+        }
+      });
+    }
+  }
+
+  private hasNoItems() {
+    return !this._card && (!this._elements || this._elements.length === 0);
```

Comment:
This is so dirty. If you are trying to create a generic conditional card, why is there knowledge in here about cards or elements? That's not generic at all.